### PR TITLE
Slider/range slider: validate properties

### PR DESCRIPTION
**Pull request by jobvs**

### CODE REVIEW

File: packages/slider/src/Slider.tsx [hunk 1]
```diff
@@ -18,31 +21,42 @@ export class Slider extends Component<Props, State> {
...
+        const max = toNumber(this.props.maximumValue);
+        const step = toNumber(this.props.stepSize);
+
+        const invalidPropsMessage =
+            value != null && min != null && max != null && step != null ? validate(min, max, step, value) : undefined;
+
+        const editable =
+            this.props.editable !== "never" &&
+            !this.props.valueAttribute.readOnly &&
+            invalidPropsMessage == null &&
```
Comment (Andries-Smit) [inline packages/slider/src/Slider.tsx]: null or undefined?

File: packages/slider/src/Slider.tsx [hunk 2]
```diff
@@ -18,31 +21,42 @@ export class Slider extends Component<Props, State> {
...
-        const enabled = this.props.editable !== "never" && !this.props.valueAttribute.readOnly;
-        const step =
-            this.props.stepSize.value && this.props.stepSize.value.gt(0) ? Number(this.props.stepSize.value) : 1;
+        const value = toNumber(this.props.valueAttribute);
+        const min = toNumber(this.props.minimumValue);
+        const max = toNumber(this.props.maximumValue);
+        const step = toNumber(this.props.stepSize);
+
+        const invalidPropsMessage =
+            value != null && min != null && max != null && step != null ? validate(min, max, step, value) : undefined;
```
Comment (Andries-Smit) [inline packages/slider/src/Slider.tsx]: toNumber returns `number | undefined`

File: packages/slider/src/Slider.tsx [hunk 3]
```diff
@@ -54,9 +68,7 @@ export class Slider extends Component<Props, State> {
     }
 
     private onSlide(values: number[]): void {
-        if (this.props.valueAttribute.status === ValueStatus.Available) {
-            this.props.valueAttribute.setTextValue(String(values[0]));
-        }
+        this.props.valueAttribute.setTextValue(String(values[0]));
```
Comment (Andries-Smit) [inline packages/slider/src/Slider.tsx]: Why cast to string?

File: packages/slider/src/Slider.tsx (hunk 2 above)
Comment (Andries-Smit) [inline packages/slider/src/Slider.tsx]: Should we show an error message too when one of the min, max, step or value is empty? Though there is a case that they all are.... when there is not context, that in it self is not an error state.. maybe we can derive that info from the dynamicValue state ValueStatus.Unavailable

We should add a test case validating what happen is the user does not have read right on of the expression values...
And test if on of the EditableValues are not writable and or readable

File: packages/slider/src/Slider.tsx (hunk 1 above)
Comment (jobvs) [reply packages/slider/src/Slider.tsx]: `x == null` is a shorthand for `x === null || x === undefined`

File: packages/slider/src/Slider.tsx (hunk 2 above)
Comment (jobvs) [reply packages/slider/src/Slider.tsx]: Same shorthand here

File: packages/slider/src/Slider.tsx (hunk 3 above)
Comment (jobvs) [reply packages/slider/src/Slider.tsx]: This was a workaround at some point, should indeed be changed to `setValue(new Big(...))` now, will do this in a separate story.

File: packages/slider/src/Slider.tsx (hunk 2 above)
Comment (jobvs) [reply packages/slider/src/Slider.tsx]: Updated to show validation messages when expression or attribute values are unavailable, and show multiple messages at the same time.

File: packages/range-slider/src/RangeSlider.tsx [hunk 4]
```diff
@@ -99,4 +95,73 @@ export class RangeSlider extends Component<Props, State> {
...
+            minimumValue.status === ValueStatus.Available &&
+            maximumValue.status === ValueStatus.Available &&
+            minimumValue.value.gt(maximumValue.value)
+        ) {
+            messages.push("The minimum value can not be greater than the maximum value.");
+        }
+        if (stepSize.status === ValueStatus.Available && stepSize.value.lte(0)) {
+            messages.push("The step size can not be zero or less than zero.");
+        }
+        if (
```
Comment (Andries-Smit) [inline packages/range-slider/src/RangeSlider.tsx]: This code block add once and if statement, all are vars are available, then validate the correct values

File: packages/range-slider/src/RangeSlider.tsx [hunk 5]
```diff
@@ -60,6 +63,9 @@ export class RangeSlider extends Component<Props, State> {
                 {this.props.upperValueAttribute.validation && (
                     <Text style={this.styles.validationMessage}>{this.props.upperValueAttribute.validation}</Text>
                 )}
+                {validationMessages.length > 0 && (
+                    <Text style={this.styles.validationMessage}>{validationMessages.join(" ")}</Text>
```
Comment (Andries-Smit) [inline packages/range-slider/src/RangeSlider.tsx]: can they be joint with a newline?

File: packages/range-slider/src/RangeSlider.tsx (hunk 4 above)
Comment (jobvs) [reply packages/range-slider/src/RangeSlider.tsx]: Done

File: packages/range-slider/src/RangeSlider.tsx (hunk 5 above)
Comment (jobvs) [reply packages/range-slider/src/RangeSlider.tsx]: Done

**Pull request merged**